tasks/publish-pyxis-repository/publish-pyxis-repository.yaml

```diff
@@ -95,13 +95,24 @@ spec:
             PYXIS_REPOSITORY=${PYXIS_REPOSITORY//----//}
             PYXIS_REPOSITORY_JSON=$(curl --retry 5 --key /tmp/key --cert /tmp/crt \
                 "${PYXIS_URL}/v1/repositories/registry/${PYXIS_REGISTRY}/repository/${PYXIS_REPOSITORY}" -X GET)
+        
             PYXIS_REPOSITORY_ID=$(jq -r '._id // ""' <<< $PYXIS_REPOSITORY_JSON)
             if [ -z "$PYXIS_REPOSITORY_ID" ]; then
                 echo Error: Unable to get Container Repository object id from Pyxis
                 echo "Pyxis response for ${PYXIS_REGISTRY}/${PYXIS_REPOSITORY}:"
                 echo $PYXIS_REPOSITORY_JSON
                 exit 1
             fi
+
+            # verify that publish_on_push is set to true.
+            # otherwise, do not publish the image.
+            PYXIS_REPOSITORY_PUBLISH_ON_PUSH=$(jq -r '.publish_on_push // ""' <<< $PYXIS_REPOSITORY_JSON)
```

Comment:
Maybe it should default to `"false"` instead of `""`.

Reply:
fixed! thanks